Cython/Utility/AsyncGen.c

```diff
@@ -216,15 +218,22 @@ __Pyx_async_gen_init_hooks(__pyx_PyAsyncGenObject *o)
 
     o->ag_hooks_inited = 1;
 
+#if CYTHON_COMPILING_IN_CPYTHON
     tstate = __Pyx_PyThreadState_Current;
-
     finalizer = tstate->async_gen_finalizer;
+#else
+    finalizer = _PyEval_GetAsyncGenFinalizer();
+#endif
     if (finalizer) {
         Py_INCREF(finalizer);
         o->ag_finalizer = finalizer;
     }
 
+#if CYTHON_COMPILING_IN_CPYTHON
     firstiter = tstate->async_gen_firstiter;
+#else
+    firstiter = _PyEval_GetAsyncGenFirstiter();
+#endif
```

Comment:
It seems these are only needed in the `test_async_gen_asyncio_gc_aclose_09` test I skipped. I skipped it since the cython version adds a finalizer to `tp_dealloc` which revives the object, which PyPy refuses to do and `exit()`'s instead. In the pure python version somehow that does not happen and the test fails, but does not crash. Adding a `gc.collect()` after the `del gen` to the pure python version makes the test pass.

Reply:
No, these are needed in more tests. Modified the github action to use the PyPy3.9 nightly that contains these functions.